tests/Unit/PointwiseFunctions/AnalyticData/CurvedWaveEquation/Test_ScalarWaveGr.cpp

```diff
@@ -228,6 +243,36 @@ void test_construct_from_options() {
     test_kerr(curved_wave_data_constructed_from_opts, bh_solution,
               flat_wave_solution);
   }
+  {  // test with wave profile = PureSphericalHarmonic
+    const auto curved_wave_data_constructed_from_opts =
+        TestHelpers::test_creation<
+            CurvedScalarWave::AnalyticData::ScalarWaveGr<
+                CurvedScalarWave::AnalyticData::PureSphericalHarmonic,
+                gr::Solutions::KerrSchild>,
+            Metavariables>(
+            "Background:\n"
+            "  Mass: 0.7\n"
+            "  Spin: [0.1, -0.2, 0.3]\n"
+            "  Center: [0.7,0.6,-2.]\n"
+            "ScalarField:\n"
+            "  Radius: 10.\n"
+            "  Width: 1.\n"
+            "  Mode: [4, -3]");
+
+    // construct internals of data constructed from options
+    const gr::Solutions::KerrSchild bh_solution(0.7, {{0.1, -0.2, 0.3}},
+                                                {{0.7, 0.6, -2.}});
+    const CurvedScalarWave::AnalyticData::PureSphericalHarmonic wave_solution{
+        10., 1., {4, -3}};
+
```

Comment:
[optional] consider adding `const auto curved_scalar_wave_data = serialize_and_deserialize(..._from_opts);` and using that in the `CHECK` and the `test_kerr` call to verify serialization is working.